test/fork/LidoFixedPriceMultiLpARM/SwapTokensForExactTokens.t.sol

```diff
@@ -383,20 +427,56 @@ contract Fork_Concrete_LidoARM_SwapTokensForExactTokens_Test is Fork_Shared_Test
             address(this) // to
         );
 
-        // State after
-        uint256 balanceWETHAfterThis = weth.balanceOf(address(this));
-        uint256 balanceSTETHAfterThis = steth.balanceOf(address(this));
-        uint256 balanceWETHAfterARM = weth.balanceOf(address(lidoARM));
-        uint256 balanceSTETHAfterARM = steth.balanceOf(address(lidoARM));
-
         // Assertions
-        assertEq(balanceWETHBeforeThis + amountOut, balanceWETHAfterThis, "WETH user balance");
+        assertGe(lidoARM.totalAssets(), totalAssetsBefore, "total assets after");
+        assertEq(weth.balanceOf(address(this)), balanceWETHBeforeThis + amountOut, "WETH user balance");
         assertApproxEqAbs(
-            balanceSTETHBeforeThis, balanceSTETHAfterThis + amountIn, STETH_ERROR_ROUNDING, "STETH user balance"
+            steth.balanceOf(address(this)),
+            balanceSTETHBeforeThis - amountIn,
+            STETH_ERROR_ROUNDING,
+            "STETH user balance"
         );
-        assertEq(balanceWETHBeforeARM, balanceWETHAfterARM + amountOut, "WETH ARM balance");
+        assertEq(weth.balanceOf(address(lidoARM)), balanceWETHBeforeARM - amountOut, "WETH ARM balance");
         assertApproxEqAbs(
-            balanceSTETHBeforeARM + amountIn, balanceSTETHAfterARM, STETH_ERROR_ROUNDING, "STETH ARM balance"
+            steth.balanceOf(address(lidoARM)),
+            balanceSTETHBeforeARM + amountIn,
+            STETH_ERROR_ROUNDING,
+            "STETH ARM balance"
         );
     }
+
+    /// @notice If the buy and sell prices are very close together and the stETH transferred into
+    /// the ARM is truncated, then there should be enough rounding protection against losing total assets.
+    function test_SwapTokensForExactTokens_Steth_Transfer_Truncated()
```

Comment:
As this is a classic test (not a fuzz), maybe we can put it under the "Passing test" category?